ols/app/models/config.py

```diff
@@ -445,6 +481,12 @@ def __init__(self, data: Optional[dict] = None):
         v = data.get("OLSConfig")
         if v is not None:
             self.ols_config = OLSConfig(v)
+        v = data.get("Authentication")
+        if v is not None:
+            self.authentication = AuthenticationConfig(v)
+        else:
+            self.authentication = AuthenticationConfig({})
```

Comment:
we need to ensure that authentication is only turned off explicitly and deliberately.  We do not want someone to accidentally have a typo in their configuration which results in no authentication happening.

see my other comment about how i added an explicit developer-mode config setting to disable auth.  If that setting isn't enable(set to true), then missing/invalid authentication configuration should be an immediate error and failure.